pkg/metrics/metrics.go

```diff
@@ -0,0 +1,98 @@
+package metrics
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Error to be  returned when marketplace is unable to reach quay.io.
+type OpsrcError string
+
+const (
+	// Default values which are to be used as labels
+	// for prometheus metrics.
+	Error     = "opsrc_error"
+	OpsrcName = "opsrc_name"
+
+	// Path, Port and Host where custom metrics would be exposed.
+	MetricsPath = "/metrics"
+	MetricsPort = "8383"
```

Comment:
This should just be an `integer`

Reply:
MetricsPort as an integer cannot to passed to `http.ListenAndServe`. Casting is required in either one of the place (before starting the server, or before passing it to `ExposeMetrics` function)